crates/steel-parser/src/lexer.rs

```diff
@@ -455,6 +509,42 @@ impl<'a> Lexer<'a> {
     }
 }
 
+struct IdentBuffer<'a> {
```

Comment:
This is a small helper to optimize all those `Cow`s. We only allocate a buffer for the identifier if we encounter an explicit escape.